### PR TITLE
Fixed datepicker keyboard input

### PR DESCRIPTION
**Pull request by shkreios**
Draft version of potential solution to #91

### CODE REVIEW

File: packages/rhf-mui/src/DatePickerElement.tsx [hunk 1]
```diff
@@ -63,62 +68,43 @@ export default function DatePickerElement<TFieldValues extends FieldValues>({
...
-              } else {
-                newValue = value
-              }
-            } else {
-              if (typeof parseDate === 'function') {
-                newValue = parseDate(value)
-              } else {
-                newValue = value
-              }
-            }
```
Comment (shkreios) [inline packages/rhf-mui/src/DatePickerElement.tsx:89]: Simplified into
```ts
const newValue =
  typeof parseDate === 'function'
    ? parseDate(value, keyboardInputValue)
    : value
```

File: packages/rhf-mui/src/DatePickerElement.tsx [hunk 2]
```diff
@@ -63,62 +68,43 @@ export default function DatePickerElement<TFieldValues extends FieldValues>({
...
+          {...mergeProps(rest, {
+            value: value,
+            onClose: onBlur,
```
Comment (shkreios) [inline packages/rhf-mui/src/DatePickerElement.tsx:73]: `mergeProps` merges method props by calling them in order.

File: packages/rhf-mui/src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    // skip merging undefined values
+    value === undefined ||
+    // skip if both value are the same primitive value
+    (typeof value !== 'object' && value === target[key])
+  ) {
+    return
+  } else if (!(key in target)) {
+    target[key] = value
+  } else {
+    mergeProps(target[key], value)
```
Comment (shkreios) [inline packages/rhf-mui/src/utils.ts:30]: Inline copy of [`merge-props`](https://github.com/andrewbranch/merge-props/blob/master/src/index.ts) to allow deep nested props

**Comment by dohomi** [comment]
I attempt to fix the bug with latest version

**Comment by dohomi** [comment]
I attempt to fix the bug with latest version

**Pull request closed without merging**